lib/engine/tile.rb

```diff
@@ -96,6 +96,12 @@ def self.part(type, params, cache)
         junction = Part::Junction.new
         cache << junction
         junction
+      when 'icon'
+        Part::Icon.new(params['image'],
+                       params.fetch('name', params['image']),
```

Comment:
let's not do anything fancy here, do it in the initializer, so just raw params['name'], params['sticky']

then in intiailizer

@name = name || image etc

also sticky should be a boolean, so it should be real truthy value